src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -77,41 +91,295 @@
     @In
     private NUIManager nuiManager;
 
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private PrefabManager prefabManager;
+
+    @In
+    private DelayManager delayManager;
+
     private static final Logger logger = LoggerFactory.getLogger(BreedingSystem.class);
     private Random random = new FastRandom();
 
     /**
-     * Receives the BreedGooeyEvent when the "activated" gooey is chosen for breeding.
+     * This method is called when a gooey steps on a breeding block
+     *
+     * @param event     The OnEnterBlockEvent
+     * @param entity    the gooey entity
+     */
+    @ReceiveEvent
+    public void onEnterBlock(OnEnterBlockEvent event, EntityRef entity, GooeyComponent gooeyComponent) {
+        LocationComponent loc = entity.getComponent(LocationComponent.class);
+        Vector3f pos = loc.getWorldPosition();
```

Comment:
It may make it clearer if you convert to `Vector2i` here, rather than in the `BlockEntityManager` call below